src/client/components/scatter_plot/model.ts

```diff
@@ -0,0 +1,139 @@
+import { action, observable } from 'mobx'
+import * as Plotly from 'plotly.js'
+
+export class ScatterplotModel {
+
+  @observable public fps: number
+  @observable public maxPoints: number
+  @observable public graphType: string
+
+  public traces: Map<string, Trace>
+  public graphUpdateX: Map<number, number[]>
+  public graphUpdateY: Map<number, number[]>
+  public graphUpdateZ: Map<number, number[]>
+  public plotlyCanvas: HTMLDivElement
+  private nextTraceId: number = 0
+
+  public constructor(opts: ScatterplotModelProps) {
+    Object.assign(this, opts)
+  }
+
+  public static of() {
+    return new ScatterplotModel({
+      fps: 30,
+      maxPoints: 100,
+      traces: new Map<string, Trace>(),
+      graphUpdateX: new Map<number, number[]>(),
+      graphUpdateY: new Map<number, number[]>(),
+      graphUpdateZ: new Map<number, number[]>(),
+      graphType: 'scatterplotgl',
+    })
+  }
+
+  @action
```

Comment:
I haven't had a chance yet to think about this enough to give a proper suggestion, but do you think its possible to keep all `@action` annotations within the controller? This lets us keep consistent with having all model mutations within controllers.

(p.s. `network.ts` is an pseudo-exception to this rule, since it is essentially a network-only controller just for reacting to network events, rather than browser events).